gui/job-details.lua

```diff
@@ -0,0 +1,438 @@
+        if dfhack.job.isSuitableItem(iobj, i, -1) then
+            count = count + 1
+            if count > 1 then return nil end
+            itype = i
+        end
+    end
+
+    return itype
+end
+
+function JobDetails:onChangeMat()
+    local idx, obj = self.list:getSelected()
+
+    if obj.iobj.item_type == -1 and obj.iobj.mat_type == -1 then
+        -- If the job allows only one specific item type, use it
+        local vitype = self:findUnambiguousItem(obj.iobj)
+
+        if vitype then
+            obj.iobj.item_type = vitype
+        else
+            dlg.showMessage(
+                'Bug Alert',
+                { 'Please set a specific item type first.\n\n',
+                  'Otherwise the material will be matched\n',
+                  'incorrectly due to a limitation in DF code.' },
+                COLOR_YELLOW
+            )
+            return
+        end
+    end
+
+    guimat.MaterialDialog{
+        prompt = 'Please select a new material for input '..idx,
+        none_caption = 'any material',
+        mat_filter = function(mat,parent,mat_type,mat_index)
+            return dfhack.job.isSuitableMaterial(obj.iobj, mat_type, mat_index, obj.iobj.item_type)
+        end,
+        on_select = self:callback('setMaterial', obj)
+    }:show()
+end
+
+function JobDetails:canChangeTrait()
+    local idx, obj = self.list:getSelected()
+    return obj ~= nil and not is_caste_mat(obj.iobj)
+end
+
+function JobDetails:onChangeTrait()
+    local idx, obj = self.list:getSelected()
+    guimat.ItemTraitsDialog{
+        job_item = obj.iobj,
+        prompt = 'Please select traits for input '..idx,
+        none_caption = 'no traits',
+    }:show()
+end
+
+local function ScrJobDetails()
+    return df.global.game.main_interface.job_details
+end
+
+local function show_job_details()
+    local scr = ScrJobDetails()
+    if not scr.open -- dfhack.gui.matchFocusString('dwarfmode/JobDetails')
+    then
+        qerror("This script needs to be run from a job details screen")
+    end
+
+    local job
+    if scr.context == df.job_details_context_type.BUILDING_TASK_LIST then
+        job = scr.jb
+    elseif scr.context == df.job_details_context_type.MANAGER_WORK_ORDER then
+        job = scr.wq
+    end
+    if job == nil then
+        qerror("Unhandled screen context: ".. df.job_details_context_type[scr.context])
+    end
+
+    local dlg = JobDetails{ job = job, context = scr.context }
+    dlg:show()
+end
+
+-- --------------------
+-- DetailsHotkeyOverlay
+--
+
+local focusStrings = 'dwarfmode/JobDetails'
+
+DetailsHotkeyOverlay = defclass(DetailsHotkeyOverlay, overlay.OverlayWidget)
+DetailsHotkeyOverlay.ATTRS{
+    default_pos={x=0,y=0},
+    default_enabled=true,
+    viewscreens=focusStrings,
+    frame={w=1+6+2+(7)+1, h=3},
+}
+
+function DetailsHotkeyOverlay:init()
+    self:addviews{
+        widgets.TextButton{
+            view_id = 'button',
+            frame={t=0, l=0, r=0, h=1},
+            label='details',
```

Comment:
Be sure to include the word "DFHack" in here somewhere so it's clear it's not a vanilla widget. Maybe 'DFHack adjust job details'?

Reply:
Wouldn't this make the buttons (unnecessary) wide...

I was trying to stay in style with this:
![image](https://github.com/DFHack/scripts/assets/348150/6f556cd2-2f43-4cca-be8f-67bfdc2b58c0)

And the pop-up window has `DFHack` on it:

![image](https://github.com/DFHack/scripts/assets/348150/4b5a8d8c-36eb-4294-bdcc-28ebf8cde341)

It's not hard to change, I'm just not quite sure we need to.

Reply:
It's a fair question. The red brackets perhaps are distinctive enough.

Instead of `details`, though, maybe `Configure job inputs`? "Details" might be too generic.